### DISCUSSION TITLE
error: multiple input filenames provided on windows

### DISCUSSION THREAD

**Discussion by HanSolo13** [bbqsrc/cargo-ndk#88] [Q&A] [answered]
Hello,
I am pretty noob, doing this as hobby, so I apologize for my question as it is likely a dumb one to which the answer is obvious.
But any help is appriciated as I cannot find out the solution alone:

I wanted to use rust with flutter, using flutter_rust_bridge package, and following a tutorial I ended up trying to use cargo-ndk on my windows machine.  But anyhow I try to use it (e.g. using the given example), I always get an error like this:

[2023-02-10T06:52:35Z INFO  cargo_ndk::cli] Using NDK at path: C:\Users\MyUserName\AppData\Local\Android\sdk\ndk\25.2.9519653 (Standard Location)
[2023-02-10T06:52:35Z INFO  cargo_ndk::cli] NDK API level: 21
[2023-02-10T06:52:35Z INFO  cargo_ndk::cli] Building targets: armeabi-v7a, arm64-v8a
[2023-02-10T06:52:35Z INFO  cargo_ndk::cli] Building armeabi-v7a (armv7-linux-androideabi)
error: failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names -Ld:\01 Private - MyUserName\01 development playground\04 Flutter\practice_my_app\rust\target\cargo-ndk\libgcc-workaround --target armv7-linux-androideabi --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=split-debuginfo --print=crate-name --print=cfg` (exit code: 1)
  --- stderr
  error: multiple input filenames provided (first two filenames are `-` and `Private`)

[2023-02-10T06:52:35Z INFO  cargo_ndk::cli] If the build failed due to a missing target, you can run this command:
[2023-02-10T06:52:35Z INFO  cargo_ndk::cli]
[2023-02-10T06:52:35Z INFO  cargo_ndk::cli]     rustup target install armv7-linux-androideabi

It seems to me, that there's an issue with the -L flag, but I cannot find out what should I do.
Thank you for any reply,
Sandor

**Comment by HanSolo13** [comment, marked as answer]
Ok, I found out that project location cannot contain any whitespaces on Windows. It was not obvious for me as nor Flutter, nor Rust needs that on their own.

**Reply by MarijnS95** [reply]
In my understanding the specific bit of code responsible for this is https://github.com/bbqsrc/cargo-ndk/commit/1be6fc5eb57dd332d298431f66307aecdc6bb4a8#r97866408; by applying the solution provided in that comment `cargo-ndk` should once again support project paths with spaces.

**Reply by HanSolo13** [reply]
Thanks for your answer, I appreciate it! As a mere user I am happy to see flutter_rust_bridge finally working. Considering my speed and level of knowledge, changing path was (a bit annoying but) acceptable solution in my case. Anyway, thanks for the guidence, now I know where to look at if I really want to solve the problem and go under the hood!
